### PR TITLE
Arm TrapezoidProfileCommand and drive practice changes

### PR DESCRIPTION
**Pull request by loafdog**
Started out as changes to get trapezoid profile for arm movement working.  Added lots of changes after Saturday drive practice as well.

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -40,7 +42,7 @@ public class RobotContainer {
   public final RobotConfig robotConfig;
   private static final String robotNameKey = "Robot Name";
   private static GenericEntry ampModeEntry = null;
-  private static boolean ampMode = false;
+  private static boolean ampMode = true;
```
Comment (BrownGenius) [inline src/main/java/frc/robot/RobotContainer.java]: Why would we want ampMode to be the default?

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -147,22 +149,48 @@ private void sysIdToShuffleboard() {
...
         .y()
-        .onTrue(
-            new InstantCommand(
-                () -> {
-                  ampMode = !ampMode;
-                  if (ampModeEntry != null) {
-                    ampModeEntry.setBoolean(ampMode);
-                  }
-                }));
+        .onTrue(new PrepareForScore(RobotConfig.arm, RobotConfig.shooter, () -> !ampMode));
```
Comment (BrownGenius) [inline src/main/java/frc/robot/RobotContainer.java:172]: Please update https://docs.google.com/document/d/1A22hMpvmXuujwkwF299oZQ94aNbzZQalUD1ifmwRZtk with any changes to the controls before pushing to baseline.

File: src/main/java/frc/robot/config/RobotConfig.java [hunk 3]
```diff
@@ -32,22 +32,36 @@ public static class DriveConstants {
...
+    public static double pidMaxOutput = 0.4;
+    public static double pidMinOutput = -0.4;
+
     public static double ffKs = 0.0;
     public static double ffKg = 0.1;
     public static double ffKv = 0.0;
     public static double ffKa = 0.0;
 
+    public static double maxVelocity = 3.0;
+    public static double maxAcceleration = 6.0;
```
Comment (BrownGenius) [inline src/main/java/frc/robot/config/RobotConfig.java:53]: We should add units to these

File: src/main/java/frc/robot/subsystems/arm/ArmIOSparkMax.java [hunk 4]
```diff
@@ -34,64 +29,75 @@ public ArmIOSparkMax(int id) {
   public ArmIOSparkMax(int id, boolean inverted) {
     /* Instantiate 1x SparkMax motors and absolute encoder */
     motor = new CANSparkMax(id, MotorType.kBrushless);
-    encoder = motor.getEncoder();
-    leftPid = motor.getPIDController();
+
+    // first thing we do to spark device is reset it to known defaults.
+    motor.restoreFactoryDefaults();
```
Comment (BrownGenius) [inline src/main/java/frc/robot/subsystems/arm/ArmIOSparkMax.java:34]: Should we do this for ALL sparkmax IO implementations?

**Review by BrownGenius — DISMISSED** [review]
Looks good.  I didn't review the subsystem specific changes to commands/subsystems.  The only main thing is to make sure the controls document is updated accordingly.

File: src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (loafdog) [reply src/main/java/frc/robot/RobotContainer.java]: no idea... i will revert this.

File: src/main/java/frc/robot/subsystems/arm/ArmIOSparkMax.java (hunk 4 above)
Comment (loafdog) [reply src/main/java/frc/robot/subsystems/arm/ArmIOSparkMax.java:34]: Yes, I think so.

File: src/main/java/frc/robot/RobotContainer.java (hunk 2 above)
Comment (loafdog) [reply src/main/java/frc/robot/RobotContainer.java:172]: Done, updated Basic Dev/Debug Controls section.

File: src/main/java/frc/robot/config/RobotConfig.java (hunk 3 above)
Comment (loafdog) [reply src/main/java/frc/robot/config/RobotConfig.java:53]: I'll be honest I don't know what the units would be.. deg/s, rad/s, rotations/s, m/s, etc... I'd rather leave them off than be wrong.

File: src/main/java/frc/robot/subsystems/arm/ArmIOSparkMax.java (hunk 4 above)
Comment (loafdog) [reply src/main/java/frc/robot/subsystems/arm/ArmIOSparkMax.java:34]: I updated for sparkmax subsystem IO files.  I did not change intakeIOTalonSRX

**Review by BrownGenius — APPROVED** [review]

**Pull request merged**